### PR TITLE
Rate us dialog and dimes adjusted

### PR DESCRIPTION
**Pull request by vivekkjha**

### CODE REVIEW

File: app/src/main/java/com/vvorld/speed/common/utils/InternetUtil.java [hunk 1]
```diff
@@ -0,0 +1,18 @@
+package com.vvorld.speed.common.utils;
```
Comment (vaibhav4444) [inline app/src/main/java/com/vvorld/speed/common/utils/InternetUtil.java]: Please shift this method in FunctionUtils class, creating separate class will only increase apk size.

File: app/src/main/java/com/vvorld/speed/framework/view/fragments/BaseFragment.java [hunk 2]
```diff
@@ -88,4 +93,17 @@ public void setIsFragmentVisible(boolean isVisible) {
...
 
+    /**
+     * Show rate us dialog checking flag of never show again, internet conenctivity and count
+     */
+    protected void showRateUsDialog()
+    {
+        int countBeforeRateUs = SpeedApplication.getPrefs().getCountBeforeRateUs();
+        if(!SpeedApplication.getPrefs().isNeverShowAgainClicked()
+                && InternetUtil.isNetworkAvailable(mActivity)
+                && countBeforeRateUs >2 )
```
Comment (vaibhav4444) [inline app/src/main/java/com/vvorld/speed/framework/view/fragments/BaseFragment.java]: Please create a constant for it & use.

File: app/src/main/res/layout/fragment_speed_layout.xml [hunk 3]
```diff
@@ -50,7 +50,7 @@
                 android:layout_height="wrap_content"
                 android:text="0"
                 android:textSize="@dimen/txtkphMphSize"
-                android:visibility="gone"/>
+                android:visibility="visible"/>
```
Comment (vaibhav4444) [inline app/src/main/res/layout/fragment_speed_layout.xml]: set it to gone as default.

File: app/src/main/res/values/strings.xml [hunk 4]
```diff
@@ -40,5 +40,11 @@
     <string name="gps_permission_rationale">We need GPS coordinates to aptly find the speed of flight and to provide you a seamless experience.\nContinue to grant location permission."</string>
     <string name="never_ask_again_denied">You have denied location permission of app. Please enable it to make app work.</string>
     <string name="goto_settings">Go to settings</string>
+    <string name="rate_app">Rate flight speed</string>
+    <string name="rate_app_content">If you enjoy using the app, would you mind taking a moment to rate it? Thank you for your support.</string>
+    <string name="rate_now">Rate now</string>
+    <string name="no_thanks">No, thanks</string>
+    <string name="remind_later">Later</string>
+    <string name="dont_show_again">Don\'t show this again.</string>
```
Comment (vaibhav4444) [inline app/src/main/res/values/strings.xml]: remove this string if it is not used.

File: app/src/main/res/layout/fragment_speed_layout.xml (hunk 3 above)
Comment (vivekkjha) [inline app/src/main/res/layout/fragment_speed_layout.xml]: Please don't set it to gone ...
Animation won't work properly then

Get Outlook for Android<https://aka.ms/ghei36>



On Thu, Aug 25, 2016 at 2:55 PM +0530, "Vaibhav" <notifications@github.com<mailto:notifications@github.com>> wrote:


In app/src/main/res/layout/fragment_speed_layout.xml<https://github.com/vaibhav4444/Speed/pull/45#discussion_r76209232>:

> @@ -50,7 +50,7 @@
>                  android:layout_height="wrap_content"
>                  android:text="0"
>                  android:textSize="@dimen/txtkphMphSize"
> -                android:visibility="gone"/>
> +                android:visibility="visible"/>


set it to gone as default.

-

**Pull request merged**